rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/action/TbCopyAttributesToEntityViewNodeTest.java

```diff
@@ -0,0 +1,287 @@
+            return newMsg;
+        }).when(ctxMock).newMsg(any(), any(String.class), any(), any(), any(), any());
+
+        node.onMsg(ctxMock, msg);
+
+        verify(entityViewServiceMock).findEntityViewsByTenantIdAndEntityIdAsync(eq(TENANT_ID), eq(DEVICE_ID));
+        verify(telemetryServiceMock).saveAndNotify(eq(TENANT_ID), eq(entityViewId), eq(AttributeScope.CLIENT_SCOPE), eq(Collections.emptyList()), any(FutureCallback.class));
+        verify(ctxMock).ack(eq(msg));
+        TbMsg expectedNewMsg = TbMsg.newMsg(msg.getQueueName(), msg.getType(), entityViewId, msg.getCustomerId(), msg.getMetaData(), msg.getData());
+        verify(ctxMock).enqueueForTellNext(refEq(expectedNewMsg, "ts", "id", "ctx"), eq("Success"));
+    }
+
+    @Test
+    public void givenAttributesValidityPeriodOutOfStartDateAndEndDate_whenOnMsg_thenThrowsException() {
+        EntityViewId entityViewId = EntityViewId.fromString("d117f1a4-24ea-4fdd-b94e-5a472e99d925");
+        EntityView entityView = new EntityView(entityViewId);
+        entityView.setStartTimeMs(Instant.now().minus(2, ChronoUnit.DAYS).toEpochMilli());
+        entityView.setStartTimeMs(Instant.now().minus(1, ChronoUnit.DAYS).toEpochMilli());
+        AttributesEntityView attributes = new AttributesEntityView(List.of("attribute1"), Collections.emptyList(), Collections.emptyList());
+        entityView.setKeys(new TelemetryEntityView(Collections.emptyList(), attributes));
+
+        when(ctxMock.getEntityViewService()).thenReturn(entityViewServiceMock);
+        when(ctxMock.getTenantId()).thenReturn(TENANT_ID);
+        when(entityViewServiceMock.findEntityViewsByTenantIdAndEntityIdAsync(any(), any()))
+                .thenReturn(Futures.immediateFuture(List.of(entityView)));
+
+        TbMsg msg = TbMsg.newMsg(
+                ATTRIBUTES_DELETED, DEVICE_ID, new TbMsgMetaData(Map.of("scope", "CLIENT_SCOPE")),
+                "{\"attributes\": [\"attribute1\"]}");
+
+        node.onMsg(ctxMock, msg);
+
+        ArgumentCaptor<Throwable> captor = ArgumentCaptor.forClass(Throwable.class);
+        verify(ctxMock).tellFailure(eq(msg), captor.capture());
+    }
+
+    @Test
+    public void givenEmptyMetadata_whenOnMsg_thenThrowsException() {
+        TbMsg msg = TbMsg.newMsg(
+                ATTRIBUTES_UPDATED, DEVICE_ID, TbMsgMetaData.EMPTY, TbMsg.EMPTY_JSON_OBJECT);
+
+        node.onMsg(ctxMock, msg);
+
+        ArgumentCaptor<Throwable> captor = ArgumentCaptor.forClass(Throwable.class);
+        verify(ctxMock).tellFailure(eq(msg), captor.capture());
+        Throwable throwable = captor.getValue();
+        assertThat(throwable.getClass()).isEqualTo(IllegalArgumentException.class);
+        String expectedExceptionMessage = "java.lang.IllegalArgumentException: Message metadata is empty";
+        assertThat(throwable.toString()).isEqualTo(expectedExceptionMessage);
```

Comment:
use: 

        assertThat(throwable).isInstanceOf(IllegalArgumentException.class).hasMessage("Message metadata is empty");